cpp/examples/advanced/cugraph_operations/graph_operations.cu

```diff
@@ -0,0 +1,253 @@
+
+  auto gpu_row_comm_size = static_cast<int>(sqrt(static_cast<double>(comm_size)));
+  while (comm_size % gpu_row_comm_size != 0) {
+    --gpu_row_comm_size;
+  }
+
+  cugraph::partition_manager::init_subcomm(*handle, gpu_row_comm_size);
+
+  return std::move(handle);
+}
+
+/**
+ * @brief This function reads graph from an input csv file and
+ * display vertex and edge partitions.
+ */
+
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
+void perform_example_graph_operations(raft::handle_t const& handle,
+                                      std::string const& csv_graph_file_path,
+                                      const bool weighted = false)
+{
+  auto const comm_rank = handle.get_comms().get_rank();
+  auto const comm_size = handle.get_comms().get_size();
+
+  std::cout << "Rank_" << comm_rank << ", reading graph from " << csv_graph_file_path << std::endl;
+
+  bool renumber = true;  // must be true for distributed graph.
+
+  // Read a graph (along with edge properties e.g. edge weights, if provided) from
+  // the input csv file
+
+  auto [graph, edge_weights, renumber_map] =
+    cugraph::test::read_graph_from_csv_file<vertex_t, edge_t, weight_t, false, multi_gpu>(
+      handle, csv_graph_file_path, weighted, renumber);
+
+  // Non-owning view of the graph object
+  auto graph_view = graph.view();
+  // Number of vertices mapped to this process, ie the size of
+  // the vertex partition assigned to this process
+
+  vertex_t size_of_the_vertex_partition_assigned_to_this_process =
+    graph_view.local_vertex_partition_range_size();
+
+  // Non-owning of the edge edge_weights object
+  auto edge_weight_view = edge_weights ? std::make_optional((*edge_weights).view()) : std::nullopt;
+
+  using graph_view_t = cugraph::graph_view_t<vertex_t, edge_t, false, multi_gpu>;
+
+  //
+  // As an example operation, compute the weighted average of the properties of
+  // neighboring vertices, weighted by the edge weights, if the input graph is weighted;
+  // Otherwise, compute the simple average.
+  //
+  if (weighted) {
+    using result_t      = weight_t;
+    auto vertex_weights = compute_out_weight_sums(handle, graph_view, *edge_weight_view);
+
+    cugraph::edge_src_property_t<graph_view_t, result_t> src_vertex_weights_cache(handle,
+                                                                                  graph_view);
+
+    cugraph::edge_dst_property_t<graph_view_t, result_t> dst_vertex_weights_cache(handle,
+                                                                                  graph_view);
+
+    update_edge_src_property(handle, graph_view, vertex_weights.begin(), src_vertex_weights_cache);
+
+    update_edge_dst_property(handle, graph_view, vertex_weights.begin(), dst_vertex_weights_cache);
+
+    rmm::device_uvector<result_t> outputs(size_of_the_vertex_partition_assigned_to_this_process,
+                                          handle.get_stream());
```

Comment:
I think more descriptive name (or at least a comment saying what this variable stores considering that this is for novice users) will be helpful, you are summing each vertex's destination neighbor's out-weight_sum * edge_weight, right?

Reply:
changed.